diff --git a/src/globalshortcutsregistry.cpp b/src/globalshortcutsregistry.cpp
index 3cbaded..ddbde7d 100644
--- a/src/globalshortcutsregistry.cpp
+++ b/src/globalshortcutsregistry.cpp
@@ -259,7 +259,7 @@ GlobalShortcutsRegistry::GlobalShortcutsRegistry()
         _manager->setEnabled(true);
     }
 
-    connect(KSycoca::self(), &KSycoca::databaseChanged, this, &GlobalShortcutsRegistry::refreshServices);
+    // connect(KSycoca::self(), &KSycoca::databaseChanged, this, &GlobalShortcutsRegistry::refreshServices);
 }
 
 GlobalShortcutsRegistry::~GlobalShortcutsRegistry()
