diff --git a/Telegram/SourceFiles/data/data_channel.cpp b/Telegram/SourceFiles/data/data_channel.cpp
index 157a16b39..8d998a1da 100644
--- a/Telegram/SourceFiles/data/data_channel.cpp
+++ b/Telegram/SourceFiles/data/data_channel.cpp
@@ -892,7 +892,7 @@ void ChannelData::clearInvitePeek() {
 }
 
 TimeId ChannelData::invitePeekExpires() const {
-	return _invitePeek ? _invitePeek->expires : 0;
+	return 0;
 }
 
 QString ChannelData::invitePeekHash() const {
